direct/tpm2/policy_test.go

```diff
@@ -281,6 +282,189 @@ func TestPolicyOrUpdate(t *testing.T) {
 	}
 }
 
+func TestPolicyPCRUpdate(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	// Use a trial session to calculate this policy
+	sess, cleanup2, err := PolicySession(thetpm, tpm.AlgSHA1, 16, Trial())
+	if err != nil {
+		t.Fatalf("setting up policy session: %v", err)
+	}
+	defer func() {
+		t.Helper()
+		if err := cleanup2(); err != nil {
+			t.Errorf("cleaning up policy session: %v", err)
+		}
+	}()
+
+	PCRs, err := CreatePCRSelection([]int{0, 1, 2, 3, 7})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+	selection := tpml.PCRSelection{
+		PCRSelections: []tpms.PCRSelection{
+			{
+				Hash:      tpm.AlgSHA1,
+				PCRSelect: PCRs,
+			},
+		},
+	}
+
+	pcrRead := PCRRead{
+		PCRSelectionIn: selection,
+	}
+
+	pcrReadRsp, err := pcrRead.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("Failed to read PCRSelection")
+	}
+
+	var expectedVal []byte
+	for _, digest := range pcrReadRsp.PCRValues.Digests {
+		expectedVal = append(expectedVal, digest.Buffer...)
+	}
+	t.Logf("expectedVal=%x", expectedVal)
+
+	// Hash algorithm must match the one in PolicySession.
+	expectedDigest := sha1.Sum(expectedVal)
+	t.Logf("expectedDigest=%x", expectedDigest)
+
+	wrongCase := append(expectedVal, []byte("wrongCase")...)
+	wrongDigest := sha1.Sum(wrongCase)
+
+	tests := []struct {
+		desc           string
+		expectedDigest []byte
+		expectedResult bool
+	}{
+		{desc: "success: expectedDigest passed in expectedDigest", expectedDigest: expectedDigest[:], expectedResult: true},
+		{desc: "fail: wrongDigest passed in expectedDigest", expectedDigest: wrongDigest[:], expectedResult: false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			policyPCR := PolicyPCR{
+				PolicySession: sess.Handle(),
+				PcrDigest: tpm2b.Digest{
+					Buffer: tt.expectedDigest,
+				},
+				Pcrs: selection,
+			}
+
+			if err := policyPCR.Execute(thetpm); err != nil {
+				t.Fatalf("executing PolicyPCR: %v", err)
+			}
+
+			pgd := PolicyGetDigest{
+				PolicySession: sess.Handle(),
+			}
+			want, err := pgd.Execute(thetpm)
+			if err != nil {
+				t.Fatalf("executing PolicyGetDigest: %v", err)
+			}
+
+			// Use the policy helper to calculate the same policy
+			pol, err := NewPolicyCalculator(tpm.AlgSHA1)
+			if err != nil {
+				t.Fatalf("creating policy calculator: %v", err)
+			}
+			policyPCR.Update(pol)
+			got := pol.Hash()
+
+			if bytes.Equal(got.Digest, want.PolicyDigest.Buffer) != tt.expectedResult {
+				t.Errorf("policyPCR.Hash() = %x,\nwant %x", got.Digest, want.PolicyDigest.Buffer)
+			}
+		})
+	}
+}
+
+// Testing void input for PcrDigest, from TPM 2.0 Part 3, 23.7.
```

Comment:
@chrisfenner, do you know in practice how often TPMs follow:
"If the caller does not
provide PCR settings (pcrDigest has a length of zero), the TPM may (and it is preferred to) use the
current TPM PCR settings (digestTPM) in the calculation for the new policyDigest."
vs "may return an error"?

Reply:
Great question. I actually learned that TPMs did this because of reviewing this change - I think it is safer and more normal to always specify the hash. That said, go-tpm should let the user do what they want, and if they want to use this feature (and rely on a TPM doing it) we should behave properly. Since the reference code does it, it seems good to me to test it.

Reply:
Yeah this was more for my curiosity. No change needed here